ui/raidboss/data/05-shb/raid/e8s.js

```diff
@@ -317,30 +317,201 @@
+            let extra = '';
+            if (data.rampant.chainSwapTank!=='Stay') {
+              if (data.rampant.chain.chains.tank.source == data.me)
+                extra = ', Swap '+data.rampant.chain.chainSwapTank;
+              else if (data.rampant.chainSwapDps == data.me)
+                extra = ', Swap NW';
+              else if (data.party.isHealer(data.me))
+                extra = ', Tank swap with '+data.rampant.chain.chainSwapTank;
+            }
+            ret = {
+              en: 'Chain on YOU'+extra,
+              de: 'Kette auf DIR',
+              fr: 'Chaine sur VOUS',
+              ko: '사슬 대상자',
+              cn: '连线',
+            };
+          }
+          delete data.rampant;
+          return ret;
+        }
       },
     },
     {
       id: 'E8S Holy Light',
       regex: Regexes.tether({ id: '0002' }),
       condition: Conditions.targetIsYou(),
-      infoText: {
-        en: 'Orb on YOU',
-        de: 'Orb auf DIR',
-        fr: 'Orbe sur VOUS',
-        ko: '구슬 대상자',
-        cn: '拉球',
+      promise: function(data, matches) {
+        if (data.options.cactbote8sLightRampantStrat === undefined || data.options.cactbote8sLightRampantStrat === 'none')
+          return null;
+
+        let p = new Promise(async (res) => {
+          let combatantData = await window.callOverlayHandler({
+            call: 'getCombatants',
+            ids: [matches.sourceId],
+          });
+          data.rampant = data.rampant || {};
+          if (!(combatantData !== null &&
+            combatantData.combatants &&
+            combatantData.combatants.length)) {
+            data.rampant.orb = null;
+            res();
+            return;
+          }
+          data.rampant.orb = combatantData.combatants.pop();
+          res();
+        });
+        return p;
+      },
+      infoText: function(data) {
+        if (data.options.cactbote8sLightRampantStrat === undefined ||
+          data.options.cactbote8sLightRampantStrat === 'none' ||
+          data.rampantOrb === null) {
+          return {
+            en: 'Orb on YOU',
+            de: 'Orb auf DIR',
+            fr: 'Orbe sur VOUS',
+            ko: '구슬 대상자',
+            cn: '拉球',
+          };
+        }
+
+        return {
+          en: 'Orb on YOU, '+data.rampantOrbText,
+          de: 'Orb auf DIR, '+data.rampantOrbText,
+          fr: 'Orbe sur VOUS, '+data.rampantOrbText,
+          ko: '구슬 대상자, '+data.rampantOrbText,
+          cn: '拉球, '+data.rampantOrbText,
+        };
+      },
+      run: function(data, matches) {
+        if (data.options.cactbote8sLightRampantStrat === undefined ||
+          data.options.cactbote8sLightRampantStrat === 'none' ||
+          data.rampantOrb === null)
+          return;
+
+        if (data.options.cactbote8sLightRampantStrat === 'sharingan') {
+          // Take orb to opposite side of spawn
+          if (data.rampantOrb.PosX > 105)
+            data.rampantOrbText = 'W';
```

Comment:
I think 'W' on its own is maybe unclear.  I think also this makes translations tricky to append raw text like this.  Could this section be combined with infoText above, and maybe do something akin to:

```
if (no strat) {
  return { orb on you }
} else if (data.rampantOrb.PosX > 105) {
  return { bait orb W }
} else if (etc) {
} else {
  return { Bait Orb (???) }
}
```

Reply:
I've reworked this but I'm not sure how to handle the translations. Should I just leave other languages out, or default it to the "orb on you" translation? Or run the text through google translate for the languages that were already present for the "orb on you" text?